app/queries/servers/system.ts

```diff
@@ -543,3 +543,41 @@ export const observeLastServerVersionCheck = (database: Database) => {
         switchMap((model) => of$(parseInt(model.value, 10))),
     );
 };
+
+export const observeIfHighlightWithoutNotificationHasLicense = (database: Database) => {
+    const IsLicensed = observeLicense(database).pipe(
+        switchMap((l) => of$(l?.IsLicensed === 'true')),
+        distinctUntilChanged(),
+    );
+
+    const isCloudStarterFree = observeLicense(database).pipe(
+        switchMap((l) => {
+            const isCloud = l?.Cloud === 'true';
+            const isStarterSKU = l?.SkuShortName === License.SKU_SHORT_NAME.Starter;
+
+            return of$(isCloud && isStarterSKU);
+        }),
+        distinctUntilChanged(),
+    );
+
+    const isStarterSKULicense = observeLicense(database).pipe(
+        switchMap((l) => {
+            const isLicensed = l?.IsLicensed === 'true';
+            const isSelfHostedStarterProduct = l?.SelfHostedProducts === License.SelfHostedProducts.STARTER;
+
+            return of$(isLicensed && isSelfHostedStarterProduct);
+        }),
+        distinctUntilChanged(),
+    );
+
+    const isEnterpriseReady = observeConfigBooleanValue(database, 'BuildEnterpriseReady', false);
+
+    return combineLatest([isCloudStarterFree, isStarterSKULicense, isEnterpriseReady, IsLicensed]).pipe(
+        switchMap(([isCSF, isSKL, isEnt, isLic]) => {
+            const isSelfHostedStarter = isEnt && isLic === false;
+            const isEnterpriseOrCloudOrSKUStarterFree = isCSF || isSKL || isSelfHostedStarter;
+
+            return of$(!isEnterpriseOrCloudOrSKUStarterFree && isEnt);
```

Comment:
It is a bit hard to follow all the conditions, but I cannot think of a better way to make them clearer.

Maybe some comments about what exactly are you checking? (i.e. what are the kind of licenses that have it and what not)